src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmNameServiceImpl.java

```diff
@@ -23,33 +31,67 @@
 	@Autowired
 	private GermplasmNameRequestValidator germplasmNameRequestValidator;
 
+	@Autowired
+	private GermplasmCodeNameBatchRequestValidator germplasmCodeNameBatchRequestValidator;
+
 	@Autowired
 	private LocationValidator locationValidator;
 
 	@Autowired
 	private SecurityService securityService;
 
+	@Autowired
+	private GermplasmCodeGenerationService germplasmCodeGenerationService;
+
+	@Autowired
+	private GermplasmNamingService germplasmNamingService;
+
 	@Override
 	public void deleteName(final Integer gid, final Integer nameId) {
 		this.germplasmNameRequestValidator.validateNameDeletable(gid, nameId);
-		germplasmNameService.deleteName(nameId);
+		this.germplasmNameService.deleteName(nameId);
 	}
 
 	@Override
-	public void updateName(final String programUUID, final GermplasmNameRequestDto germplasmNameRequestDto, final Integer gid, final Integer nameId) {
+	public void updateName(final String programUUID, final GermplasmNameRequestDto germplasmNameRequestDto, final Integer gid,
+		final Integer nameId) {
 		final BindingResult errors = new MapBindingResult(new HashMap<String, String>(), GermplasmNameRequestDto.class.getName());
-		germplasmNameRequestValidator.validate(programUUID, germplasmNameRequestDto, gid, nameId);
+		this.germplasmNameRequestValidator.validate(programUUID, germplasmNameRequestDto, gid, nameId);
 		if (germplasmNameRequestDto.getLocationId() != null) {
-			locationValidator.validateLocation(errors, germplasmNameRequestDto.getLocationId(), programUUID);
+			this.locationValidator.validateLocation(errors, germplasmNameRequestDto.getLocationId(), programUUID);
 		}
-		germplasmNameService.updateName(germplasmNameRequestDto, gid, nameId);
+		this.germplasmNameService.updateName(germplasmNameRequestDto, gid, nameId);
 	}
 
 	@Override
 	public Integer createName(final String programUUID, final GermplasmNameRequestDto germplasmNameRequestDto, final Integer gid) {
 		final BindingResult errors = new MapBindingResult(new HashMap<String, String>(), GermplasmNameRequestDto.class.getName());
-		germplasmNameRequestValidator.validate(programUUID, germplasmNameRequestDto, gid, null);
-		locationValidator.validateLocation(errors, germplasmNameRequestDto.getLocationId(), programUUID);
-		return germplasmNameService.createName(germplasmNameRequestDto, gid);
+		this.germplasmNameRequestValidator.validate(programUUID, germplasmNameRequestDto, gid, null);
+		this.locationValidator.validateLocation(errors, germplasmNameRequestDto.getLocationId(), programUUID);
+		return this.germplasmNameService.createName(germplasmNameRequestDto, gid);
+	}
+
+	@Override
+	public List<GermplasmGroupNamingResult> createCodeNames(final GermplasmCodeNameBatchRequestDto germplasmCodeNameBatchRequestDto) {
+		this.germplasmCodeNameBatchRequestValidator.validate(germplasmCodeNameBatchRequestDto);
+		try {
```

Comment:
try-catch here doesn't is unnecessary for me

Reply:
Those are checked exceptions so we need to handle it. I just moved the try catch inside GermplasmCodeGenerationService.

Reply:
Done